File: iosApp/iosApp/screens/permissions_required/LocationPermissionsDeniedScreen.swift
import SwiftUI
import KMPObservableViewModelSwiftUI
import Shared

struct LocationPermissionsDeniedScreen: View {
    var body: some View {
        VStack {
            Text("You have denied location permissions")
                .foregroundStyle(Color(.primary))
                .font(.system(size:20, weight: .bold))
                .padding()
                .multilineTextAlignment(.center)
            
            Spacer().frame(height: 16)

            Text("This app requires location permissions to wake you up when you're near your destination")
                .foregroundStyle(Color(.primary))
                .padding()
                .multilineTextAlignment(.center)
            
            Spacer().frame(height: 16)

            Text("Please go to settings to grant location permissions to this app")
                .foregroundStyle(Color(.primary))
                .padding()
                .multilineTextAlignment(.center)
        }
    }
}


#Preview("LocationDenied") {
    LocationPermissionsDeniedScreen()
}

File: composeApp/src/androidMain/kotlin/uk/co/oliverdelange/location_alarm/screens/permissions/LocationPermissionsDeniedScreen.kt
package uk.co.oliverdelange.location_alarm.screens.permissions

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

@Composable
fun LocationPermissionsDeniedScreen() {
    Column(
        Modifier
            .fillMaxSize()
            .padding(40.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Text(
            "You have denied location permissions",
            color = MaterialTheme.colorScheme.primary,
            textAlign = TextAlign.Center,
            fontWeight = FontWeight.Bold
        )
        Spacer(Modifier.height(16.dp))
        Text(
            "This app requires location permissions to wake you up when you're near your destination",
            color = MaterialTheme.colorScheme.primary,
            textAlign = TextAlign.Center
        )
        Spacer(Modifier.height(16.dp))
        Text(
            "Please go to settings to grant location permissions to this app",
            color = MaterialTheme.colorScheme.primary,
            textAlign = TextAlign.Center
        )
    }
}